File: src/app/componentes/modelos/informacion/informacion.ts
export class Informacion {
}

// src/app/modelos/informacion.ts
export interface InfoCotizacionRead {
  cotizacionesid: number;
  flo_precioUnitarioCotizar?: number | null;
  flo_precioTotalCotizar?: number | null;
  flo_diferencia?: number | null;
  flo_precioUnitarioBase?: number | null;
  precio_venta?: number | null;
  items_id?: number;
  int_orden?: number;
}
// --- Primero el ItemRead, que usamos dentro de InformacionRead ---
export interface ItemRead {
  items_id: number;
  txt_cpc: string;
  txt_unidad: string;
  txt_especificaciones: string | null;
  int_cantidad: number;
  flo_precioUnitario: number;
  flo_precioTotal: number;
  flo_total: number;
  proforma_id: number;
  txt_evidencia?: string | null;  // ✅ Agregado campo evidencia
  cotizaciones?: InfoCotizacionRead[];
}

// --- Payload para crear una Informacion (no incluye proforma_id ni items) ---
export interface InformacionCreate {
  txt_cliente?: string;
  txt_ruc?: string;
  txt_direccion?: string;
  txt_fecha?: string;   // o Date, según prefieras
  txt_telefono?: string;
  txt_necesidad?: string;
  txt_funcionario?: string;
  txt_correo?: string;
  tHora_maxina?: string;
  txt_objetivoCompra?: string;
  txt_plazoEntrega?: string;
  txt_vigenciaOferta?: string;
  txt_formaPago?: string;
  txt_metodologiaTrabajo?: string;
  txt_garantia?: string;
  txt_enlace?: string;

}

// --- Lo que recibes al leer/crear (incluye proforma_id y el array items) ---
export interface InformacionRead extends InformacionCreate {
  tTimeHora: any;
  dFechaRegistro: any;
  proforma_id: number;
  estado_id: number;      // <-- añade esto
  estado_name: string;      // <-- y esto
  items: ItemRead[];
  txt_enlace?: string;

}

/**
 * Respuesta de GET /informaciones/ 
 * (array completo de cabeceras con sus items anidados)
 */
export interface InformacionListResponse {
  count: number;
  next: string | null;
  previous: string | null;
  results: InformacionRead[];
}

File: app/schemasFolder/informacion.py
# app/schemas/informacion.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import date, time

from app.schemasFolder.items import ItemCreate, ItemCreateIn, ItemRead



class InformacionBase(BaseModel):
    txt_cliente:        Optional[str] = None
    txt_ruc:            Optional[str] = None
    txt_direccion:      Optional[str] = None
    txt_fecha:          Optional[str]  = None
    txt_telefono:       Optional[str] = None
    txt_necesidad:      Optional[str] = None
    txt_funcionario:    Optional[str] = None
    txt_correo:         Optional[str] = None
    tHora_maxina:       Optional[str] = None
    txt_objetivoCompra: Optional[str] = None
    txt_plazoEntrega : Optional[str] = None
    txt_vigenciaOferta   : Optional[str] = None
    txt_garantia : Optional[str] = None
    txt_formaPago : Optional[str] = None
    txt_metodologiaTrabajo : Optional[str] = None
    txt_enlace : Optional[str] = None
    txt_infimaNro : Optional[str] = None
    estado_id:      Optional[int]   = None
    dFechaRegistro:  Optional[date] = None
    tTimeHora:       Optional[time] = None
    txtUsuarioRegistra : Optional[str] = None
    txt_numeroProforma: Optional[str] = None

class InformacionCreate(InformacionBase):
    pass

class InformacionRead(InformacionBase):
    proforma_id: int
    estado_name: str
    dFechaRegistro: date
    tTimeHora:    time
    items:       List[ItemRead] = []

    class Config:
        orm_mode = True
class EstadoUpdate(BaseModel):
    estado_id: int

class EstadoCount(BaseModel):
    estado: str
    total: int

class ProformaReporteOut(BaseModel):
    oferente: Optional[str] = None
    proforma: Optional[str] = None
    fecha_proforma: Optional[str] = None
    codigo_proceso: Optional[str] = None
    entidad_contratante: Optional[str] = None
    objeto_compra: Optional[str] = None
    valor_contrato: Optional[float] = None
    plazo_contractual: Optional[str] = None

class InformacionCreateWithItems(InformacionCreate):
    items: List[ItemCreateIn]